specification/dsdl/dsdl.tex

```diff
@@ -132,30 +132,77 @@ \section{Syntax}\label{sec:dsdl_syntax}
 directives apply only to the part of the service type definition where they are defined,
 not crossing the boundary of the service response marker.
 
-\subsection{Attribute definition}
 
-Field definitions follow one of the below specified declaration patterns:
+\subsection{Constant expressions}\label{sec:dsdl_constant_expressions}
+A valid constant expression is either just a constant, or an expression that results in a constant.
+Such expression is given in infix notation,
+this means \texttt{A <op> B} for binary operators and \texttt{<op> A} for unary operators.
+A comprehensive table of legal operations on types is defined below,
+the order of the table also defines the order of precedence between operations.
+
+\begin{UAVCANSimpleTable}{Operators for constant expressions}{|c c c c c l|} \label{table:dsdl_const_operators}
+Presedence & Op & A & B & Res & Description \\
+1 & \texttt{!}    &  \texttt{const bool}   & N/A                  & \texttt{const bool}    & Logical not \\
```

Comment:
Should we not use `not` instead? We are not encumbered with C legacy, and it makes sense to optimize for readability because DSDL types are write-once, read-anywhere kind of thing?

Reply:
The argument for using symbols over words is: it's pretty clear that symbols are not identifiers
 and If we want to add more at a later point, this can be done without reserving keywords prematurely.

Most modern language (that I'm aware of) still uses `!`, `&&` and `||`. The best reason to go against this would be to do whatever Pythons doing. But, Python being Python, replacing the symbols with the words should be trivial. i almost expect the following statement to parse `replace "!" with "not" in DSDL`.

Reply:
I mean is a question of design. We should not design things to match our existing implementations. Surely we can find a way to evaluate an expression in Python, regardless of the chosen syntax. Shall we vote?

Reply:
> Shall we vote

Absolutely, +1 for symbols (`&&`, `||` and `!`) to be in line with C like syntax (used by: C/C++, C#, Java, Rust, etc) and, more importantly, to keep the number of reserved keywords low.

Reply:
C++ allows using not, or, and, etc instead of symbols (https://en.cppreference.com/w/cpp/language/operator_alternative) so this isn't just the pythonic way of doing things. I'm for words if it doesn't impeded RUST compatibility.

Reply:
I also like words. Even if we don't reserve `not`, `or`, and friends, they still can't be safely used because they are likely to be reserved in the target language (C/C++/Python/Ada/Verilog).